core/src/main/java/hudson/cli/DisconnectNodeCommand.java

```diff
@@ -0,0 +1,106 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.Computer;
+import hudson.util.EditDistance;
+import jenkins.model.Jenkins;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * @author pjanouse
+ * @since TODO
+ */
+@Extension
+public class DisconnectNodeCommand extends CLICommand {
+    @Argument(metaVar = "NAME", usage = "Slave name, or empty string for master; comama-separated list is supported", required = true, multiValued = true)
+    private List<String> nodes;
+
+    @Option(name = "-m", usage = "Record the note about why you are disconnecting this node")
+    public String cause;
+
+    private static final Logger LOGGER = Logger.getLogger(DisconnectNodeCommand.class.getName());
+
+    @Override
+    public String getShortDescription() {
+
+        return Messages.DisconnectNodeCommand_ShortDescription();
+    }
+
+    @Override
+    protected int run() throws Exception {
+        boolean errorOccurred = false;
+        final Jenkins jenkins = Jenkins.getActiveInstance();
+
+        final HashSet<String> hs = new HashSet<String>();
+        hs.addAll(nodes);
+
+        List<String> names = null;
+
+        for (String node_s : hs) {
+            Computer computer = null;
+
+            try {
+                computer = jenkins.getComputer(node_s);
+
+                if (computer == null) {
+                    if (names == null) {
+                        names = new ArrayList<String>();
+                        for (Computer c : jenkins.getComputers())
+                            if (!c.getName().isEmpty())
+                                names.add(c.getName());
+                    }
+                    String adv = EditDistance.findNearest(node_s, names);
+                    throw new IllegalArgumentException(adv == null ?
+                            hudson.model.Messages.Computer_NoSuchSlaveExistsWithoutAdvice(node_s) :
+                            hudson.model.Messages.Computer_NoSuchSlaveExists(node_s, adv));
+                }
+
+                computer.cliDisconnect(cause);
+            } catch (Exception e) {
+                if (hs.size() == 1) {
+                    throw e;
+                }
+
+                final String errorMsg = String.format(node_s + ": " + e.getMessage());
+                stderr.println(errorMsg);
+                errorOccurred = true;
+                continue;
+            }
+        }
+
+        if (errorOccurred) {
```

Comment:
this will not be shown if you supply just one computername.  
Why would/should this behave differently for one vs many computers (is this the common pattern for CLIs)?